db/db_universal_compaction_test.cc

```diff
@@ -2230,6 +2231,209 @@ TEST_F(DBTestUniversalCompaction2, PeriodicCompaction) {
   ASSERT_EQ(4, output_level);
 }
 
+TEST_F(DBTestUniversalCompaction2, PeriodicCompactionOffpeak) {
+  constexpr int kSecondsPerDay = 86400;
+  constexpr int kSecondsPerHour = 3600;
+  constexpr int kSecondsPerMinute = 60;
+
+  Options opts = CurrentOptions();
+  opts.compaction_style = kCompactionStyleUniversal;
+  opts.level0_file_num_compaction_trigger = 10;
+  opts.max_open_files = -1;
+  opts.compaction_options_universal.size_ratio = 10;
+  opts.compaction_options_universal.min_merge_width = 2;
+  opts.compaction_options_universal.max_size_amplification_percent = 200;
+  opts.periodic_compaction_seconds = 5 * kSecondsPerDay;  // 5 days
+  opts.num_levels = 5;
+
+  // Just to add some extra random days to current time
+  Random rnd(test::RandomSeed());
+  int days = rnd.Uniform(100);
+
+  int periodic_compactions = 0;
```

Comment:
Probably will need atomic_int.

Reply:
Doesn't compaction picking happen while holding the DB mutex? If so, there would only be a race with the test code if there's a race in executing the periodic compactions.

Reply:
Yes, compaction picking happens while holding the db mutex. I was worried that the test code can access `periodic_compactions` while compaction picking is happening. That seems to be not the case. I guess if TSAN complains about it, it's fine too and it means the test has a bug.

Reply:
I will leave it without `atomic` and see what TSAN says.